<?xml version="1.0" ?>
<launch>
  <arg name="robot" default="panda"/>
  <arg name="robot_ip" default="172.16.0.2"/>
  <arg name="arm_id" default="$(arg robot)" />
  <include file="$(find franka_control)/launch/franka_control.launch" pass_all_args="true"/>
 <rosparam command="load" file="$(find cartesian_impedance_controller)/config/panda.yaml" subst_value="true" />
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" args="CartesianImpedance_trajectory_controller"/>
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_example_controllers)/launch/rviz/franka_description_with_marker.rviz"/>
  <node name="interactive_marker" pkg="franka_example_controllers" type="interactive_marker.py" required="true" output="screen">
    <param name="link_name" value="$(arg arm_id)_link0" />
    <remap from="equilibrium_pose" to="/CartesianImpedance_trajectory_controller/reference_pose" />
  </node>
  <node name="rqt_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" required="false" />
</launch>

